docs/paper/dev/api/components.md

```diff
@@ -0,0 +1,120 @@
+---
+slug: /dev/components
+---
+
+# Components
+
+:::note
+
+This documentation page applies to both the Paper and Velocity projects
+
+:::
+
+Since Minecraft 1.17, the game has utilized components to represent text to be displayed
+by clients. Components offer a number of benefits over plain text strings which are enumerated below.
+Paper and Velocity natively implements the Adventure API to add component support wherever possible.
+
+## Differences
+
+Previously, text was a linear structure with the only formatting options being
+confusing symbols like `§c` and `§k` to control basic colors and styles of text.
+Components are a tree-like structure which inherit style and colors from their parents.
+The docs mentioned above contain examples of this. 
+
+Components have several types which do different things than just display raw text, like
+translating text to the client's language based on a key, or showing a client-specific keybind
+to a player.
+
+All these component types support more style options like any RGB color, interaction events
+(click and hover). The other component types and these style options have poor or missing
+representations in the legacy string format.
+
+## Usage
+
+Representing text as Components is now the supported way of representing text for Paper and Velocity. They are used
+for almost all aspects of text being displayed to clients. Text like item names, lore, boss bars, team prefixes and
+suffixes, custom names, and much more all support Components in respective APIs.
+[Mojang has stated](https://bugs.mojang.com/browse/MC-190605?focusedId=993040&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-993040)
+that client support for the legacy format with `§` will be removed in the future.
+
+
+:::tip
+
+In the Paper API, there are lots of deprecated methods and types that deal with this legacy format. This is to
+signal that a better alternative in Components is available and should be migrated to going forward.
+
+:::
+
+### Components
+
+Components can be interacted with as objects. There are different interfaces for each type along with
+builders for all the types. These objects are immutable so when constructing more complex components, it's
+recommended to use builders to avoid creating new Component instances with every change.
```

Comment:
Should have an example for that, something simple like "for example, use `Component.text().content("Hi").clickEvent(something)` instead of `Component.text("Hi").clickEvent(something)`"
or a comment above the builder creation, contrasting it with a non builder usage

Reply:
Ok, I adapted the example to show both static imports and proper use of the builder.